hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HoodieCatalog.java

```diff
@@ -349,6 +350,8 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable catalogTable, boo
       final String partitions = String.join(",", resolvedTable.getPartitionKeys());
       conf.setString(FlinkOptions.PARTITION_PATH_FIELD, partitions);
       options.put(TableOptionProperties.PARTITION_COLUMNS, partitions);
+    } else {
+      conf.setString(FlinkOptions.KEYGEN_CLASS_NAME.key(), NonpartitionedAvroKeyGenerator.class.getName());
```

Comment:
The `HoodieTableFactory` would finally fix this option right?

Reply:
> The `HoodieTableFactory` would finally fix this option right?

hi,`Catalog#createTable` will write hudi conf to hoodie. properties and will not use tablefactory